packages/lodestar/src/chain/blocks/pool.ts

```diff
@@ -36,15 +28,12 @@ export class BlockPool {
       pendingBlockPool.push(job);
     } else {
       this.unknownParentBlockPool.set(key, [job]);
-      //this prevents backward syncing, tolerance is 20 blocks
-      if (job.signedBlock.message.slot <= this.forkChoice.getHead().slot + 20) {
-        const pendingBlocks = Array.from(this.unknownParentBlockPool.values()).flat();
-        this.eventBus.emit(
-          "unknownBlockRoot",
-          findUnknownAncestor(this.config, pendingBlocks, job.signedBlock.message.parentRoot)
-        );
-      }
     }
+    const pendingBlocks = Array.from(this.unknownParentBlockPool.values()).flat();
```

Comment:
@mpetrunic the main change is to emit `unknownBlockRoot` without caring about forkchoice head, logic is still the same. In the case of missing gossip blocks, sync would search for them.

Reply:
Wait but this then allows backwards sync? If we receive block far away from our head we would start syncing backwards as each time it finds missing block it would emit events again?

Reply:
yeah we allow that from regular sync time, for initial sync we don't allow backward sync.

I think from regular sync, syncing backward is a good thing bc it helps the regular sync to be finished faster: regular sync starts from our head and do `beacon_blocks_by_range` requests. At the same time gossip blocks come and they emit `unknownBlockRoot` and soon it'd overlap with following up `beacon_blocks_by_range` requests and then the node is fully synced.

Also with the latest change, we emit the missing ancestor root instead of the missing block root.